### PR TITLE
rados: fix segfaults when run without --pool

### PR DESCRIPTION
**Pull request by ddiss**
Fix and test for http://tracker.ceph.com/issues/17684 .

### CODE REVIEW

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -1898,6 +1913,11 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
   }
   if (snapid != CEPH_NOSNAP) {
     string name;
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: could you move this definition down to where it is firstly used?

**Comment by smithfarm** [comment]
`Reviewed-by: Nathan Cutler <ncutler@suse.com>`

**Comment by ddiss** [comment]
thanks for the feedback.

changes since last version
- move name string declaration down from the start of the code block
- add Nathan's reviewed-by tags

**Comment by smithfarm** [comment]
@ddiss Actually, the "Reviewed-by" tag is supposed to go into the merge commit, signifying that the PR as a whole was reviewed. So it's the reviewer who's supposed to add it when the PR is merged.

**Comment by ddiss** [comment]
@smithfarm okay, pushed again without your tags

**Comment by tchaikov** [comment]
lgtm

**Comment by ddiss** [comment]
The Jenkins TestErasureCodePluginJerasure.cc segfault doesn't appear to be in any way related to this change.

**Comment by smithfarm** [comment]
Jenkins test this please

**Comment by smithfarm** [comment]
In all likelihood, the jerasure plugin test failure is http://tracker.ceph.com/issues/17561

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2016-11-10_07:59:52-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2016-11-10_02:18:01-rados-wip-kefu-testing---basic-mira/

**Pull request merged**